contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -393,44 +384,33 @@ contract MixinStakingPoolRewards is
     /// @param poolId Unique id of pool.
     /// @param _delegatedStakeToPoolByOwner Amount of stake the member has
     ///        delegated to the pool.
-    /// @param isDependent is true iff adding a dependency. False, otherwise.
     function _setCumulativeRewardDependenciesForDelegator(
         bytes32 poolId,
-        IStructs.StoredBalance memory _delegatedStakeToPoolByOwner,
-        bool isDependent
+        IStructs.StoredBalance memory _delegatedStakeToPoolByOwner
     )
         private
     {
-        // If this delegator is not yet initialized then there's no dependency
-        // to unset.
-        if (!isDependent && !_delegatedStakeToPoolByOwner.isInitialized) {
-            return;
-        }
-
         // Get the most recent cumulative reward, which will serve as a
         // reference point when updating dependencies
-        IStructs.Fraction memory mostRecentCumulativeReward =
-            _getMostRecentCumulativeReward(poolId);
+        IStructs.Fraction memory mostRecentCumulativeReward = _getMostRecentCumulativeReward(poolId);
 
         // Record dependency on current epoch.
         if (_delegatedStakeToPoolByOwner.currentEpochBalance != 0
             || _delegatedStakeToPoolByOwner.nextEpochBalance != 0)
         {
-            _addOrRemoveDependencyOnCumulativeReward(
+            _trySetCumulativeReward(
                 poolId,
                 _delegatedStakeToPoolByOwner.currentEpoch,
-                mostRecentCumulativeReward,
-                isDependent
+                mostRecentCumulativeReward
             );
         }
 
         // Record dependency on the next epoch
```

Comment:
So one thing I was thinking about was instead of explicitly copying the CR to the next epoch, we just checked if we had a record for `epoch-1` if `epoch` was unset in `_computeMemberRewardOverInterval()`. I didn't want to do this before because I wasn't sure if reference counting would get in the way of this (because `epoch-1` could theoretically get deleted), but that shouldn't be a factor now, right?